base_layer/wallet/src/wallet.rs

```diff
@@ -309,6 +309,9 @@ where
         source_public_key: &CommsPublicKey,
         features: OutputFeatures,
         message: String,
+        sender_signature: Signature,
```

Comment:
```suggestion
        sender_metadata_signature: Signature,
```

Reply:
> One thing to note though is that you have not checked whether the receiver changed the script or the features in `add_single_recipient_info` of the sender protocol (or whenever the sender checks the recipient data). This means that the sender will only find out when the base node rejects the message, when it has the ability to fail faster.

This is done as part of `pub fn validate_internal_consistency(` when a transaction is constructed, together with 

```
        self.verify_kernel_signatures()?;
        self.validate_kernel_sum(total_offset, &factories.commitment)?;

        self.validate_range_proofs(&factories.range_proof)?;
        self.validate_sender_signatures()?;
        self.validate_script_offset(script_offset_g)
```